xclim/indices/_agro.py

```diff
@@ -1259,18 +1447,42 @@ def standardized_precipitation_evapotranspiration_index(
 
     Notes
     -----
+    If results include NaNs, check that the `offset` parameter is larger than the minimum water budget values.
+
     See Standardized Precipitation Index (SPI) for more details on usage.
     """
+    uses_default_offset = offset == "1 mm/d"
+    offset = 0 if offset is None else convert_units_to(offset, wb, context="hydro")
+    if params is not None:
+        params_offset = (
+            0
+            if params.attrs["offset"] is None
+            else convert_units_to(params.attrs["offset"], wb, context="hydro")
+        )
+        if params_offset != offset:
+            extra_msg = (
+                " (using default SPEI value 1 mm/d)" if uses_default_offset else ""
+            )
+            warnings.warn(
+                f"The offset in `params` differs from the input `offset`{extra_msg}."
+                "Procceding with the value given in `params`."
+            )
+            offset = params_offset
```

Comment:
Why this, and not the offset explicitly given ?

Reply:
This is indeed a bit more complicated than it need to be since I give "1 mm/d" as a default value. This was done to avoid doing  a breaking change. I think it might be good to have a breaking change so that users are aware of this, but if this is not the final version of SPEI, maybe we should reserve a breaking change for the final formulation?

Reply:
Your call. My suggestion is really to say that a parameter in the function call is more explicit of user intent than a attribute in another parameter, so should override the later.

Reply:
Yes, I would agree. The problem is that to avoid a breaking change, I was using a default value for `offset`. In this situation, the parameter in the function call can either be something put in explicitly by the user. Given that may be replaced in the future, a strategy could be to use a dummy value by default (e.g. "UNSET") to track if a value was indeed given.